src/api/allfinished.js

```diff
@@ -265,6 +265,83 @@ const getLeaderHistoryForLevel = async (
   return leaderHistoryWithData;
 };
 
+// checks if the current user can view times for the given
+// level and kuski.
+const authLevel = async (req, LevelIndex, KuskiIndex) => {
+  const level = await levelInfo(+LevelIndex);
+  if (!level || level.Locked) return false;
+
+  const auth = authContext(req);
+
+  if (level.Hidden) {
+    if (KuskiIndex) {
+      // user can see their own times on levels that are hidden.
+      // seems likely that user created the level in this case.
+      return auth.userid && Number(KuskiIndex) === auth.userid;
+    }
```

Comment:
if you want to see if user created the level you should compare to level.AddedBy, regardless it should be fine to see own times in any level regardless who made it, so I think it's just the comment that's misleading

Reply:
Will re look at this. Was just trying to mirror some logic from other endpoint but didn't totally make sense to me.

Reply:
I don't know the difference between locked and hidden actually, so I left the logic as is for now, ie. because I don't know what to do if the level is locked but the user owns it. Would be an easy change though.